kubernetes/structures_daemonset.go

```diff
@@ -109,6 +113,11 @@ func expandRollingUpdateDaemonSet(p []interface{}) *appsv1.RollingUpdateDaemonSe
 	}
 	in := p[0].(map[string]interface{})
 
+	if v, ok := in["max_surge"].(string); ok {
+		val := intstr.Parse(v)
```

Comment:
I am not 100% confident whether this one will work when you pass value as a percentage. I would suggest write tests first and look if it passes. In addition to that, I will try this code manually and see if I get the same result as with YAML manifests.